plugins/optimization-detective/class-od-url-metric-group-collection.php

```diff
@@ -417,26 +417,25 @@ public function get_common_lcp_element(): ?array {
 	 * (the default sample size). Therefore, given the number (n) of visited elements on the page this will only
 	 * end up running n*4*3 times.
 	 *
-	 * @todo Should there be an OD_Element class which has a $url_metric property which then in turn has a $group property. Then this would only need to return array<string, OD_Element[]>.
 	 * @since n.e.x.t
 	 *
-	 * @return array<string, non-empty-array<int, array{OD_URL_Metric_Group, OD_URL_Metric, ElementData}>> Keys are XPaths and values are arrays of tuples consisting of the group, URL metric, and element data.
+	 * @return array<string, non-empty-array<int, OD_Element>> Keys are XPaths and values are the element instances.
 	 */
-	public function get_all_denormalized_elements(): array {
+	public function get_all_elements(): array {
```

Comment:
I don't really like the `get_all_elements` name. It's more like `get_elements_grouped_by_xpath`.

Reply:
How about `get_all_elements_by_xpath()`?

Reply:
That works, but then it would seem like you could pass an `$xpath` to get just the elements with the given XPath.

Reply:
Like `document.getElementById()`

Reply:
That's fair, though I'm not sure how including the word `grouped` would clarify that. FWIW to me that `all` helps to clarify this will return "all" elements, not just the elements filtered by something.

Reply:
How about `get_xpath_elements_map()`?

Reply:
Done in 98d3fd8a